client/src/app/pages/applications/analysis-wizard/useWizardReducer.ts

```diff
@@ -0,0 +1,172 @@
+import { useCallback, useState } from "react";
+import { produce } from "immer";
+import { useImmerReducer } from "use-immer";
+
+import {
+  AdvancedOptionsState,
+  AnalysisModeState,
+  AnalysisScopeState,
+  CustomRulesStepState,
+  SetTargetsState,
+} from "./schema";
+
+export interface WizardState {
+  mode: AnalysisModeState;
+  targets: SetTargetsState;
+  scope: AnalysisScopeState;
+  customRules: CustomRulesStepState;
+  options: AdvancedOptionsState;
+  isReady: boolean;
+}
+
+const INITIAL_WIZARD_STATE: WizardState = {
+  mode: {
+    mode: "source-code-deps",
+    artifact: null,
+    isValid: false,
+  },
+  targets: {
+    selectedTargets: [],
+    selectedTargetLabels: [],
+    targetFilters: undefined,
+    isValid: true, // Targets are optional
```

Comment:
Consider separating `isValid` from `isFormValid` and/or introducing `isOptional` (or similar)
Once the user enters the step the property is populated from the form.  If the form will contain errors (for whatever reason) it will block the wizard. The user is responsible for cleaning up the state to the point that hook form will toggle `isValid`.  Restoring "empty" and "valid" form state might not always be easy or user friendly.